*** a/src/http/ngx_http_request.c        2016-03-07 00:44:08.026314251 +0300
--- b/src/http/ngx_http_request.c        2016-03-07 00:44:31.512980063 +0300
***************
*** 584,590 ****
      r->start_msec = tp->msec;

      r->method = NGX_HTTP_UNKNOWN;
-     r->http_version = NGX_HTTP_VERSION_10;

      r->headers_in.content_length_n = -1;
      r->headers_in.keep_alive_n = -1;
--- 584,589 ----
